### PR TITLE
Near2Far refactor phase 1: changes to storage and access of far fields

### PR DESCRIPTION
**Pull request by shashwat-sh**
- Near2far data structures now store E,H rather than N,L
- Significant refactoring and simplifications to how fields are accessed and re-projected to new distances

Previous behaviour:
- All data structures had a `fields` method which returned fields in a coordinate system that depended on the monitor, e.g. spherical for the angle monitor, Cartesian for the Cartesian monitor
- Radar cross section method was available for only the angle monitor
- All these methods took an argument `r` to project fields to that given distance

New behaviour
- All data structures have the following methods:
    - `fields_spherical`
    - `fields_cartesian`
    - `radar_cross_section`
    - `power`
    so that all quantities are available for all monitors, and in both coordinate systems.
- By default, fields are projected to a distance of `1m`.
- Re-projection to a different distance is no longer done with an argument. Instead, one explicitly calls `renormalize_fields(distance)` so that all fields stored in the data become renormalized to the new distance.
   - The field data structures' coordinates will be updated to remember this new distance. This way, one can re-project as many times as they want. Each time, the old phase is removed, the new phase is applied, and the field data coordinates are updated.

Rationale:
- Make things simple, explicit, and uniform for all three types of monitors.
- Unify the interface as much as possible in anticipation of exact projections, in which case everything can stay the same, except `renormalize_fields` won't be available because the phase is "baked into" the exact Green's function.

Backend PR: https://github.com/flexcompute/tidy3d/pull/557

### CODE REVIEW

File: tidy3d/components/data/monitor_data.py [hunk 1]
```diff
@@ -531,33 +531,49 @@ class FluxTimeData(MonitorData):
...
+    Hphi: PROJFIELDTYPE = pd.Field(
+        ...,
+        title="Hphi",
+        description="Spatial distribution of phi-component of the magnetic field.",
+    )
+
+    Hphi: PROJFIELDTYPE = pd.Field(
+        ...,
+        title="Hphi",
+        description="Spatial distribution of phi-component of the magnetic field.",
```
Comment (momchil-flex) [inline tidy3d/components/data/monitor_data.py:576]: This is repeated

File: tidy3d/updater.py [hunk 2]
```diff
@@ -202,6 +202,46 @@ def iterate_update_dict(update_dict: Dict, update_types: Dict[str, Callable]):
...
+        mnt_dict.pop("fields")
+        mnt_dict["proj_distance"] = 1e6
+        axis = mnt_dict.pop("u_axis")
+        mnt_dict["proj_axis"] = axis
+        return mnt_dict
+
+    iterate_update_dict(
+        update_dict=sim_dict,
+        update_types={
+            "AbstractNear2FarMonitor": fix_abstract_info,
```
Comment (momchil-flex) [inline tidy3d/updater.py:236]: Is this ever used? There should never be an abstract monitor type in the json, right?

**Review by momchil-flex — APPROVED** [review]
I think this makes a lot of sense now!

**Review by tylerflex — APPROVED** [review]

File: tidy3d/updater.py (hunk 2 above)
Comment (shashwat-sh) [reply tidy3d/updater.py:236]: Oh true, thanks.

**Pull request merged**